src/machines/mod.rs

```diff
@@ -251,6 +262,46 @@ trait Cancellable: StateMachine {
     }
 }
 
+/// We need to wrap calls to [StateMachine::on_event_mut] to track coverage, or anything else
+/// we'd like to do on every call.
+pub(crate) trait OnEventWrapper: StateMachine
+where
+    <Self as StateMachine>::State: Display,
+    <Self as StateMachine>::Event: Display,
+    Self: Clone,
+{
+    fn on_event_mut(
+        &mut self,
+        event: Self::Event,
+    ) -> Result<Vec<Self::Command>, MachineError<Self::Error>> {
+        #[cfg(test)]
+        let from_state = self.state().to_string();
+        #[cfg(test)]
+        let converted_event_str = event.to_string();
+
+        let res = StateMachine::on_event_mut(self, event);
+        if res.is_ok() {
+            #[cfg(test)]
```

Comment:
Should this be one line above and gate entire if statement?

Reply:
Yeah, in principle it probably should but in practice that makes clippy complain about this unnecessary `let` binding and this seemed an easier way to deal with that than a clippy allow. Compiler will optimize it out anyway.